### PR TITLE
Add Support for Firmware Version 1.5.5 and Update README Instructions

### PR DESCRIPTION
**Pull request by lucio97**
This pull request includes the following updates:

1. **Firmware Download Support**: Added functionality to the script to enable downloading version 1.5.5 of the firmware for the C100X.
2. **README Update**: Updated the README file to provide clearer instructions on compiling the `TcpDump2Mqtt.conf` before creating the firmware.

### CODE REVIEW

**Comment by fcaronte** [comment]
Hi, i see there is a new 1.7.3 update, do you think is possible to mod this one too?

**Comment by lucio97** [comment]
> Hi, i see there is a new 1.5.7 update, do you think is possible to mod this one too?

Hey! I checked the site, but I can't find the 1.5.7 update you're talking about. At home, I only have the 100x with DoorEntry for testing. Do you have a download link?

**Comment by fcaronte** [comment]
Actually I have stock 100x and get OTA update but can't find a link to manually download the new update. I don't know if I can flash your 1.5.5 and downgrade without problems or now I need to wait to find the new one ?

File: main.py [hunk 1]
```diff
@@ -36,6 +36,12 @@ class PrepareFirmware():
                  'liferay/bt_mxLiferayCheckout.jsp?fileFormat=generic&'
                  'fileName=C100X_010501.fwz&fileId='
                  '58107.23188.46381.34528')
+    # C100X_010505.fwz
+    url_c100x_010505 = ('https://www.homesystems-legrandgroup.com/MatrixENG/'
+                 'liferay/bt_mxLiferayCheckout.jsp?fileFormat=generic&'
+                 'fileName=C100X_010505.fwz&fileId='
+                 '58107.23188.62332.48840')
+    password = 'C300X'
```
Comment (sstidl) [inline main.py:44]: please remove the duplicate line

**Comment by sstidl** [comment]
> > Hi, i see there is a new 1.5.7 update, do you think is possible to mod this one too?
> 
> Hey! I checked the site, but I can't find the 1.5.7 update you're talking about. At home, I only have the 100x with DoorEntry for testing. Do you have a download link?

i have the update taken from the device ( `/home/bticino/cfg/extra/FW/UPGRADE.fwz` )
if you want i can share a temporary link but i also didnt find the official download link yet.

**Comment by sstidl** [comment]
> Actually I have stock 100x and get OTA update but can't find a link to manually download the new update. I don't know if I can flash your 1.5.5 and downgrade without problems or now I need to wait to find the new one ?

@fcaronte i would try it with the 1.5.5 -- i dont think there is much difference between 1.5.7 and 1.5.5

**Comment by fcaronte** [comment]
Yes but I can save Install 1.5.5 over 1.5.7? No problem with downgrade?

**Comment by lucio97** [comment]
Hi all,

I've seen in the telegram community that someone managed to perform a downgrade without issues, though I can't guarantee it works for everyone.

Regarding the current implementation, the program still requires a link even when using a local file, since the prompt appears after file selection. While a fake URL might bypass this, I believe the logic should be updated to first ask if a local file is used, then request the firmware version if opting for a download. Unfortunately, I don't have time to implement this change now.

Thanks, @sstidl, for pointing out the duplicate line at line 44!

**Comment by sstidl** [comment]
I found the URL which is used by the device to download 1.5.7:
https://eliotdmstrg.blob.core.windows.net/bs-classe100x/C100X_010507.tar.gz
In the tar file there is a .bin file which is the same as the .fwz files. 
I checked md5sum from this file and the Upgrade.fwz directly on the device (as stated in Readme)

**Comment by lomaxhh** [comment]
Hello, github is still confusing for me. Can i use this method if i have the firmware 1.5.7?

**Pull request merged**